app/src/test/java/com/example/foody/recipe_search/presentation_mvvm/SearchViewModelTest.kt

```diff
@@ -0,0 +1,189 @@
+package com.example.foody.recipe_search.presentation_mvvm
+
+import app.cash.turbine.ReceiveTurbine
+import app.cash.turbine.turbineScope
+import com.example.foody.recipe_search.domain.RecipesSearchRepository
+import com.example.foody.recipe_search.presentation_mvvm.model.RecipeListState
+import com.example.foody.recipe_search.presentation_mvvm.model.SearchScreenState
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.*
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.io.IOException
+
+class SearchViewModelTest {
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private var dispatcher = UnconfinedTestDispatcher()
```

Comment:
It is better to use `StandarTestDispatcher`. See the [link](https://craigrussell.io/2022/01/19/comparing-standardtestdispatcher-and-unconfinedtestdispatcher) I gave you the last time. `StandarTestDispatcher` gives us more fine grained control (though it does require more strict setup in tests, but that is also positive).

Reply:
Ok, so ... don't know much about either of them. But i've noticed that sometimes I have to change (or use one of dispatchers) in order for some tests to work. Some work only with Unconfined and others with Standard.

Reply:
Read the article 😉. It is good, all the details are concisely presented there.
But basically the tests which work with Unconfined could fail with Standard, but mostly not vice versa, though they can if ordering in which coroutines are launched matters.
With Standard sometimes we need to use [advanceUntilIdle()](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/-test-coroutine-scheduler/advance-until-idle.html), [runCurrent()](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/-test-coroutine-scheduler/run-current.html) or [advanceTimeBy(ms)](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/-test-coroutine-scheduler/advance-time-by.html), but that also gives us more fine grained control. Unconfined can be used when order of coroutine launches doesn't matter, but I prefer to use Standard always, because with it I am more certain that if a test passes it means all is set up as it should.